Create Docker image. Build and push to ghcr.io/mueslimak3r/tv-intro-detection:latest

Cookie-Monster-Coder opened the pull request.

Cookie-Monster-Coder commented:
idk if you want to have the `jellyfin.py` and `jellyfin_auto_skip.py` running in the same docker container or different. I think we should put the output of `jellyfin.py` in another dir called `/config` so we are able to mount it easily, while keeping all the *py in `/app'

mueslimak3r commented on README.md:
```diff
@@ -14,6 +14,23 @@ By default there is little/no output to stdout or stderr until the script has fi
+  --name=Jellyfin-intro-Scanner \
+  -e JELLYFIN_URL=http://Jellyfin:port \
+  -e JELLYFIN_USERNAME=username \
+  -e JELLYFIN_PASSWORD='password' \
```
I've added `CONFIG_DIR` and `DATA_DIR` as supported env variables.

By default, config dir is at `./config` and stores `path_map.txt` and the data dir
By default, data dir is `os.path.join(config_dir, 'data')` and it stores `fingerprints` and `jellyfin_cache`

mueslimak3r commented on README.md:
````diff
@@ -14,6 +14,23 @@ By default there is little/no output to stdout or stderr until the script has fi
 
+## Running Docker
+  ### Scanner
+  ```
````
This looks good!

At some point I'd like to have a docker compose example too, so if you want to add that it would be cool, but a docker run is also fine for now

mueslimak3r reviewed (changes requested):
Thanks for the PR!

I've pushed changes that add a "config" folder, and "data" subfolder. The location of both can be redefined with env variables. By default the data folder will inherit the location of the config folder unless otherwise defined by the env variable

I've marked this as "request changes" in case you need to change things to account for the new config and data folders

> idk if you want to have the jellyfin.py and jellyfin_auto_skip.py running in the same docker container or different.

I'm not sure yet, and I'd be okay with either separate or together. If they do run together, it's important to handle the auto skipper exiting with an error and not let that kill the library processing script.

The auto skipper still doesn't cleanly handle being disconnected from the server so it will throw an exception. The library processing script only connects to jellyfin when it starts, and then logs out after scraping the library data.

mueslimak3r commented:
fyi it's preferred if PRs are done from branches instead of master. This way it allows for easy rebasing when incorporating upstream changes. No worries here though since the only changes you've made to your fork are for this PR. But if you plan on contributing more, using a branch is preferred.

mueslimak3r commented on Dockerfile_Jellyfin-Intro-Scanner:
```diff
@@ -0,0 +1,10 @@
+RUN pip install --no-cache-dir -r /app/requirements.txt
+WORKDIR /app
+ENTRYPOINT ["python"]
+CMD [ "jellyfin.py", "-d", "-j" ]
```
I just added logging to file. with `-l` a new log is created for each session inside `/config/logs`
eg: `log_2022_03_30-10_50_59_PM.txt` 

```suggestion
CMD [ "jellyfin.py", "-d", "-l", "-j" ]
```

Cookie-Monster-Coder commented:
> fyi it's preferred if PRs are done from branches instead of master.

Sorry, I'm still new to this. Forgive me for all the spam.

I separate the Scanner and Skipper, I think that would be the best. One for the error handling as stated above and second if someone wanted to run the 'Skipper' alone or vice versa for some reason.

Added a parameter chat for docker variables.
Updated docker run commands
Added docker compose for the scanner & skipper

mueslimak3r commented on README.md:
````diff
@@ -14,6 +14,78 @@ By default there is little/no output to stdout or stderr until the script has fi
+| Required | ```-v /path/to/config:/app/config```      | Location of config/data on disk. Must use the same locations for Jellyfin-Intro-Scanner & Jellyfin-Intro-Skipper containers to work correctly together.       |
+| Required | ```-v /path/to/media/on/host:/path/to/media/on/Jellyfin/container'```      |  Location of media library on disk. If you use the same volume path for your Jellyfin container, you don't have to edit ```path_map.txt``` in your config folder. (If you need to change it you must first create a ```path_map.txt``` in your config folder. ***Not in the data subfolder***)        |
+| Optional | ```-e CONFIG_DIR=/config```      |  Pointless to change the data directory in the docker container, but it's there if needed. If you do decide to add this make sure to update your ```/app/config``` path mapping.          |
+| Optional | ```-e DATA_DIR=/config/data'```      | Pointless to change the data directory in the docker container, but it's there if needed. If you do decide to add this make sure to update your ```/app/config``` path mapping.         |
````
You're right that there isn't much use for adjusting these in the context of docker.

```suggestion
| Optional | ```-e DATA_DIR=/data'```      | Use a different directory to store cached data. Modifying this will likely require a new path mapping such as ```-v /path/to/data:/data```.         |
```

mueslimak3r commented:
> > fyi it's preferred if PRs are done from branches instead of master.
> 
> Sorry, I'm still new to this. Forgive me for all the spam.

Oh no worries, and thanks for putting the time in to set this up! It definitely wouldn't have happened for a while otherwise 🙂

> I separate the Scanner and Skipper, I think that would be the best. One for the error handling as stated above and second if someone wanted to run the 'Skipper' alone or vice versa for some reason.

> Added a parameter chat for docker variables. Updated docker run commands Added docker compose for the scanner & skipper

Looks good! I think that approach does cover the most use-cases and potential issues

Cookie-Monster-Coder commented on README.md:
````diff
@@ -14,6 +14,78 @@ By default there is little/no output to stdout or stderr until the script has fi
+| Required | ```-e JELLYFIN_PASSWORD='password'```      | Jellyfin User Password         |
+| Required | ```-v /path/to/config:/app/config```      | Location of config/data on disk. Must use the same locations for Jellyfin-Intro-Scanner & Jellyfin-Intro-Skipper containers to work correctly together.       |
+| Required | ```-v /path/to/media/on/host:/path/to/media/on/Jellyfin/container'```      |  Location of media library on disk. If you use the same volume path for your Jellyfin container, you don't have to edit ```path_map.txt``` in your config folder. (If you need to change it you must first create a ```path_map.txt``` in your config folder. ***Not in the data subfolder***)        |
+| Optional | ```-e CONFIG_DIR=/config```      |  Pointless to change the data directory in the docker container, but it's there if needed. If you do decide to add this make sure to update your ```/app/config``` path mapping.          |
````
Worded it way better than I did

Cookie-Monster-Coder commented:
BTW you need to make the two packages public.

mueslimak3r commented:
Thanks, I didn't know about that. Done!
